From 4617adc64debfde007180ca0476522823ee9e229 Mon Sep 17 00:00:00 2001
From: "Poluri, Sarat Chandra" <sarat.chandra.poluri@intel.com>
Date: Tue, 19 Feb 2019 17:43:35 -0700
Subject: [PATCH 1040/1214] media: intel-ipu4: [VIRT] Check whether file open
 was successful. If it wasn't, return error to the FE.

Without this change, anytime pipeline node is not created on the SOS,
due to errors like FW version mismatch, starting the UOS will cause
a kernel crash and cause a system reboot.

Change-Id: Ib969381265627294180bddccc0a7d607a3e579d0
Tracked-On: PKT-1760
Tracked-On: OAM-76615
Tracked-On: HSDES 1409007561
Signed-off-by: Poluri, Sarat Chandra <sarat.chandra.poluri@intel.com>
---
 drivers/media/pci/intel/virtio/intel-ipu4-virtio-be-pipeline.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/media/pci/intel/virtio/intel-ipu4-virtio-be-pipeline.c b/drivers/media/pci/intel/virtio/intel-ipu4-virtio-be-pipeline.c
index 81e2826..2297bf6 100644
--- a/drivers/media/pci/intel/virtio/intel-ipu4-virtio-be-pipeline.c
+++ b/drivers/media/pci/intel/virtio/intel-ipu4-virtio-be-pipeline.c
@@ -29,8 +29,14 @@ int process_pipeline_open(struct ipu4_virtio_req_info *req_info)
 	}
 
 	pr_info("process_device_open: /dev/intel_pipeline");
-	if (!pipeline)
+	if (!pipeline) {
 		pipeline = filp_open("/dev/intel_pipeline", O_RDWR | O_NONBLOCK, 0);
+		if (!pipeline) {
+			pr_err("%s: no pipeline device exists on host OS",
+					__func__);
+			return IPU4_REQ_ERROR;
+		}
+	}
 	guestID = domid;
 
 	return IPU4_REQ_PROCESSED;
-- 
2.7.4

